algosdk/abi.py

```diff
@@ -0,0 +1,570 @@
+        i.e. '(x,(y,(z)))' -> ['x', '(y,(z))']
+        """
+        # If the tuple content is empty, return an empty list
+        if not s:
+            return []
+
+        if s.startswith(",") or s.endswith(","):
+            raise error.ABITypeError(
+                f"cannot have leading or trailing commas in {s}"
+            )
+
+        if ",," in s:
+            raise error.ABITypeError(f"cannot have consecutive commas in {s}")
+
+        tuple_strs = []
+        depth = 0
+        word = ""
+        for char in s:
+            word += char
+            if char == "(":
+                depth += 1
+            elif char == ")":
+                depth -= 1
+            elif char == ",":
+                # If the comma is at depth 0, put the word as a new token.
+                if depth == 0:
+                    word = word[:-1]
+                    if word:
+                        tuple_strs.append(word)
+                        word = ""
+        if word:
+            tuple_strs.append(word)
+        if depth != 0:
+            raise error.ABITypeError(f"parenthesis mismatch: {s}")
+        return tuple_strs
+
+    def is_dynamic(self):
+        """
+        Returns whether the current type or any of its child types is a
+        dynamic type or not.
+        """
+        if (
+            self.abi_type_id == BaseType.ArrayDynamic
+            or self.abi_type_id == BaseType.String
+        ):
+            return True
+        for child in self.child_types:
+            if child.is_dynamic():
+                return True
+        return False
+
+    def find_bool(type_list, index, delta):
+        """
+        Helper function to find consecutive booleans from current index in a tuple.
+        """
+        until = 0
+        while True:
+            curr = index + delta * until
+            if type_list[curr].abi_type_id == BaseType.Bool:
+                if curr != len(type_list) - 1 and delta > 0:
+                    until += 1
+                elif curr > 0 and delta < 0:
+                    until += 1
+                else:
+                    break
+            else:
+                until -= 1
+                break
+        return until
+
+    def byte_len(self):
+        if self.abi_type_id == BaseType.Address:
+            return 32
+        elif self.abi_type_id == BaseType.Byte:
+            return 1
+        elif self.abi_type_id == BaseType.Bool:
+            return 1
+        elif (
+            self.abi_type_id == BaseType.Uint
+            or self.abi_type_id == BaseType.Ufixed
+        ):
+            return self.bit_size // 8
+        elif self.abi_type_id == BaseType.ArrayStatic:
+            if self.child_type.abi_type_id == BaseType.Bool:
+                # 8 Boolean values can be encoded into 1 byte
+                byte_length = self.static_length // 8
+                if self.static_length % 8 != 0:
+                    byte_length += 1
+                return byte_length
```

Comment:
There should be an easy way to do: use math.ceil, and calculate the byte_length = ceil(self.static_length / 8).
It will evaluate self.static_length / 8 to be float and ceil the result back to int.